classification/classification_network.py

```diff
@@ -246,109 +220,368 @@ def createClassifierRegion(network, classifierType, classifierParams):
+    spParams["inputWidth"] = sensorRegion.encoder.width
+    spRegion = _createSpatialPoolerRegion(network, SP_REGION_NAME, spParams)
+    _validateRegionWidths(previousRegionWidth, spRegion.getSelf().inputWidth)
+    _linkRegions(network,
+                 SENSOR_REGION_NAME,
+                 previousRegion,
+                 SP_REGION_NAME)
+    previousRegion = SP_REGION_NAME
+    previousRegionWidth = spRegion.getSelf().columnCount
+
+  # Create TM region, if enabled.
+  if networkConfiguration[TM_REGION_NAME]["enabled"]:
+    tmParams = networkConfiguration[TM_REGION_NAME]["params"]
+    tmRegion = _createTemporalMemoryRegion(network, TM_REGION_NAME, tmParams)
+    _validateRegionWidths(previousRegionWidth, tmRegion.getSelf().columnCount)
+    _linkRegions(network,
+                 SENSOR_REGION_NAME,
+                 previousRegion,
+                 TM_REGION_NAME)
+    previousRegion = TM_REGION_NAME
+    previousRegionWidth = tmRegion.getSelf().cellsPerColumn
+
+  # Create UP region, if enabled.
+  if networkConfiguration[UP_REGION_NAME]["enabled"]:
+    upParams = networkConfiguration[UP_REGION_NAME]["params"]
+    upRegion = _createUnionPoolerRegion(network, upParams)
+    # TODO: not sure about the UP region width params. This needs to be updated.
+    _validateRegionWidths(previousRegionWidth,
+                          upRegion.getSelf().cellsPerColumn)
+    _linkRegions(network,
+                 SENSOR_REGION_NAME,
+                 previousRegion,
+                 UP_REGION_NAME)
+    previousRegion = UP_REGION_NAME
+
+  # Create classifier region (always enabled)
+  classifierParams = networkConfiguration[CLASSIFIER_REGION_NAME]["params"]
+  classifierType = networkConfiguration[CLASSIFIER_REGION_NAME]["type"]
+  _createClassifierRegion(network,
+                          CLASSIFIER_REGION_NAME,
+                          classifierType,
+                          classifierParams)
+  # Link the classifier to previous region and sensor region - to send in 
+  # category labels.
+  network.link(previousRegion, CLASSIFIER_REGION_NAME, "UniformLink", "")
+  network.link(SENSOR_REGION_NAME,
+               CLASSIFIER_REGION_NAME,
+               "UniformLink",
+               "",
+               srcOutput="categoryOut",
+               destInput="categoryIn")
+
+  return network
 
-  sp = createSpatialPoolerRegion(network,
-                                 sensor.encoder.getWidth())
 
-  tm = createTemporalMemoryRegion(network)
 
-  classifier = createClassifierRegion(network,
-                                      classifierType,
-                                      classifierParams)
+def _enableRegionLearning(network,
+                          trainedRegionNames,
+                          regionName,
+                          recordNumber):
+  """
+  Enable learning for a specific region.
+  
+  @param network (Network) the network instance 
+  @param trainedRegionNames (list) regions that have been trained on the 
```

Comment:
well i use it for printing useful phase info. it's true that functionally it doesn't change anything. But i find this info useful to know what is going on in the network and how it is learning (i.e. what are the regions learning for each phase) I'd rather keep it, at least in nupic.research